include/boost/geometry/algorithms/difference.hpp

```diff
@@ -15,19 +15,21 @@
 #define BOOST_GEOMETRY_ALGORITHMS_DIFFERENCE_HPP
 
 
-#include <boost/variant/apply_visitor.hpp>
-#include <boost/variant/static_visitor.hpp>
-#include <boost/variant/variant_fwd.hpp>
-
 #include <boost/geometry/algorithms/detail/intersection/multi.hpp>
+#include <boost/geometry/algorithms/detail/make_rtree.hpp>
```

Comment:
Did we have a dependency on `rtree` in the non-index part of the library before?

Usually all the overlay algorithms use partition. Why do we need an rtree now for the same (didn't review all yet, but it's probably the same) purpose?

Reply:
Thanks for your explanation! I agree with you that code using partition is less readable.
I'm fine.